Loop/Managers/DeviceDataManager.swift

```diff
@@ -331,14 +329,32 @@ final class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverD
         // If we don't yet have pump status, or it's old, poll for it.
         if latestReservoirValue == nil || latestReservoirValue!.startDate.timeIntervalSinceNow <= -pumpStatusAgeTolerance {
             readPumpData { (result) in
+                let nsPumpStatus: NightscoutUploadKit.PumpStatus?
                 switch result {
                 case .Success(let (status, date)):
                     self.updateReservoirVolume(status.reservoir, atDate: date, withTimeLeft: nil)
+                    let battery = BatteryStatus(voltage: status.batteryVolts, status: BatteryIndicator(batteryStatus: status.batteryStatus))
+                    nsPumpStatus = NightscoutUploadKit.PumpStatus(clock: date, pumpID: status.pumpID, iob: nil, battery: battery, suspended: status.suspended, bolusing: status.bolusing, reservoir: status.reservoir)
```

Comment:
What about throwing `status` over the fence to `NightscoutUploadManager` and letting it make the decision to upload? (and store the new state on line 49)